packages/eslint-plugin-nx/src/rules/enforce-module-boundaries.ts

```diff
@@ -261,6 +360,91 @@ export default createESLintRule<Options, MessageIds>({
             data: {
               imp,
             },
+            fix(fixer) {
+              // imp is equal to @myorg/someproject
+              const indexTsPaths = getBarrelEntryPointByImportScope(imp);
+              if (indexTsPaths && indexTsPaths.length > 0) {
+                // imported JS functions to remap
+                const imports = (
+                  (node.source.parent as any).specifiers || []
+                ).map((specifier) => specifier.imported.name);
+
+                // ignore non-named imports for now, like "import '../../some-other-lib/src'"
+                if (imports.length === 0) {
+                  return;
+                }
+
+                // process each potential entry point and try to find the imports
+                const importsToRemap = [];
+
+                for (const entryPointPath of indexTsPaths) {
+                  for (const importMember of imports) {
+                    const importPath = getRelativeImportPath(
+                      importMember,
+                      entryPointPath,
+                      sourceProject.data.sourceRoot
+                    );
+                    if (importPath) {
+                      // resolve the import path
+                      let importPathResolved = relative(
+                        dirname(context.getFilename()),
+                        dirname(importPath)
+                      );
+
+                      // if the string is empty, it's the current file
+                      importPathResolved =
+                        importPathResolved === ''
+                          ? `./${basename(importPath)}`
+                          : joinPathFragments(
+                              importPathResolved,
+                              basename(importPath)
+                            );
+
+                      importsToRemap.push({
+                        member: importMember,
+                        relativePath: importPathResolved.replace('.ts', ''),
+                      });
+                    }
+                  }
+                }
+
+                // group imports from the same relative path
+                const importsToRemapGrouped = importsToRemap.reduce(
```

Comment:
Can the two `importToRemap` reducers be extracted to a runtime linter util function?

Reply:
ofc, fixing it